src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -127,14 +130,40 @@ export function initDashboardApp(app, deps) {
           $scope.create = () => {
             history.push(DashboardConstants.CREATE_NEW_DASHBOARD_URL);
           };
+          $scope.dashboardProviders = deps.dashboardProviders() || [];
+
           $scope.find = (search) => {
-            return service.find(search, $scope.listingLimit);
+            const searchRx = new RegExp(search, 'i');
+            const matchSearchToItem = (item) => {
+              const matchedTitle = !!item.title.match(searchRx);
+              const matchedDescription = !!item.description.match(searchRx);
+              const matchedType = !!item.type.match(searchRx);
+
+              return matchedTitle || matchedDescription || matchedType;
+            };
```

Comment:
searchRx is compiled from the text-input of search-bar.  "Plain text" searches work as they did in previous implementation.
Yes, the end-user _could_ use a real RX for search, though we won't advertise that and it has not been tested thoroughly.

Reply:
first of all, this is a behavior/UX change, do we have a confirmation from UX and product side to confirm it is okay to have change change in a minor version?

Meanwhile, this will require browser to load all objects in to its memory so that you can do such search. Like browser side pagination, this is not scalable either

Reply:
This change has been OK'd by UX @kgcreative 

In my testing of this feature, plugins that use HashRouter do not correctly render upon pushing the whole url using `history.push`.  The routing fails at the interface of the HashRouter, where the plugin's "default" page is rendered only.


I would very happily learn of some trick that can overcome this limitation.

Reply:
Also, all cross-plugin navigations in the app render a "full page reload" anyway.  BrowserRouter correctly detects that the "new url" is the same as "this url" and perserves as much of the browser window as it can.